shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -1282,6 +1285,20 @@ public String[] computePlatformResolvedLocale(@NonNull String[] strings) {
   }
 
   // ----- End Localization Support ----
+  @Nullable
+  public float getScaledFontSize(float fontSize, int configurationId) {
+    final DisplayMetrics metrics = SettingsChannel.getPastDisplayMetrics(configurationId);
+    if (metrics == null) {
+      Log.e(
+          TAG,
+          "getScaledFontSize called with configurationId "
+              + String.valueOf(configurationId)
+              + ", which can't be found.");
+      return -1f;
+    }
+    return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, fontSize, metrics)
+        / TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 1.0f, metrics);
```

Comment:
Alternatively, you could just use `metrics.density` instead of calling applyDimension() on 1dp: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/util/TypedValue.java;l=409?q=typedvalue